site/docs/pages/guides/migrating-to-snapchain.mdx

```diff
@@ -19,7 +19,7 @@ Snapchain is designed to be a drop-in replacement for [Hubble](https://github.co
 
 Messages once submitted must be included in blocks, similar to blockchain transactions. The `submitMessage` has two main differences from Hubble: 
 
-1. `submitMessage` requests must all contain `dataBytes` for Snapchain. The `hub-nodejs` builders handle this in all versions `>=0.13.0`, but if you're not using those you will need to manually update this like so:
+1. `submitMessage` requests can optionally include `dataBytes` for better performance. If not provided, Snapchain will automatically encode the `data` field. The `hub-nodejs` builders handle this optimization in all versions `>=0.13.0`. If you want to manually optimize this:
```

Comment:
We intentionally recommend that all clients provide dataBytes. If the client uses a different proto serialization impl than the server, we run into invalid hash in the server because client and server compute the hash off different bytes. To avoid such issues, all clients should provide dataBytes.